src/nvim/highlight.c

```diff
@@ -370,12 +370,15 @@ void update_window_hl(win_T *wp, bool invalid)
 
   // determine window specific background set in 'winhighlight'
   bool float_win = wp->w_floating && !wp->w_config.external;
-  if (float_win && hl_def[HLF_NFLOAT] != 0) {
+  if (float_win && hl_def[HLF_NFLOAT] != 0 && ns_id > 0) {
     wp->w_hl_attr_normal = hl_def[HLF_NFLOAT];
   } else if (hl_def[HLF_COUNT] > 0) {
     wp->w_hl_attr_normal = hl_def[HLF_COUNT];
+  } else if (float_win) {
+    wp->w_hl_attr_normal = HL_ATTR(HLF_NFLOAT) >
```

Comment:
more readable with `0` on this line (also allowed by formatter)

Reply:
yes. I didn't look format carefully.. just make formatc highlight.c .